weather_sp/splitter_pipeline/file_splitters.py

```diff
@@ -90,89 +85,132 @@ def should_skip(self):
 
         with self._open_grib_locally() as grbs:
             for grb in grbs:
-                key = SplitKey(grb.typeOfLevel, grb.shortName)
+                splits = dict()
+                for dim in self.split_dims:
+                    try:
+                        splits[dim] = getattr(grb, dim)
+                    except RuntimeError:
+                        self.logger.error(
+                            'Variable not found in grib: %s', dim)
+                key = self._get_output_file_path(splits)
                 if key not in outputs:
-                    metrics.Metrics.counter('file_splitters',
-                                            f'grib: {key}').inc()
                     outputs[key] = self._open_outfile(key)
                 outputs[key].write(grb.tostring())
                 outputs[key].flush()
 
             for out in outputs.values():
                 out.close()
-            self.logger.info('split %s into %d files', self.input_path, len(outputs))
+            self.logger.info('split %s into %d files',
+                             self.input_path, len(outputs))
 
     @contextmanager
     def _open_grib_locally(self) -> t.Iterator[t.Iterator[pygrib.gribmessage]]:
         with self._copy_to_local_file() as local_file:
             yield pygrib.open(local_file.name)
 
-    def _open_outfile(self, key: SplitKey):
-        return FileSystems.create(self._get_output_file_path(key))
+    def _open_outfile(self, key: str):
+        return FileSystems.create(key)
 
 
 class NetCdfSplitter(FileSplitter):
 
+    def __init__(self, input_path: str, output_info: OutFileInfo,
+                 force_split: bool = False, logging_level: int = logging.INFO):
+        super().__init__(input_path, output_info,
+                         force_split, logging_level)
+        self.output_info.set_formatting_if_needed('_{variable}')
+        self.split_dims = self._get_split_dims()
+
     def split_data(self) -> None:
+        if not self.split_dims:
+            raise ValueError('No splitting specified in template.')
         if self.should_skip():
             metrics.Metrics.counter('file_splitters', 'skipped').inc()
-            self.logger.info('Skipping %s, file already split.', repr(self.input_path))
+            self.logger.info('Skipping %s, file already split.',
+                             repr(self.input_path))
             return
-
-        with self._open_dataset_locally() as nc_data:
-            fields = [var for var in nc_data.variables.keys() if
-                      var not in nc_data.dimensions.keys()]
-            for field in fields:
-                self._create_netcdf_dataset_for_variable(nc_data, field)
-            self.logger.info('split %s into %d files', self.input_path, len(fields))
+        if any(split not in ('time', 'level', 'variable') for split in self.split_dims):
+            raise ValueError(
+                'netcdf split: unknown split dimension, supported are time, level, variable')
```

Comment:
If not now, in v2 of this feature, this should be supported.

Reply:
I'm honestly not sure how to do this, I'll have to think more and do some testing.